test/e2e/soak_k8petstore.go

```diff
@@ -0,0 +1,231 @@
+		strconv.Itoa(restServers),
+		strconv.Itoa(loadGenerators),
+		"1", // number of redis slaves.
+		"0", // dont bother embedded test.
+		ns,
+	)
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	Logf("Starting k8petstore application....")
+	//Run the k8petstore app, and log / fail if it returns any errors.
+	//This should return quickly, assuming containers are downloaded.
+	if err = cmd.Start(); err != nil {
+		log.Fatal(err)
+	}
+	//Make sure exit code != 0
+	if err = cmd.Wait(); err != nil {
+		if exiterr, ok := err.(*exec.ExitError); ok {
+			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
+				log.Printf("Exit Status: %d", status.ExitStatus())
+			}
+		}
+	}
+	Expect(err).NotTo(HaveOccurred())
+	Logf("... Done starting k8petstore ")
+
+	//Lets poll until we reach min_expected transactions
+	totalTransactions := 0
+	startupTimeoutSeconds := 500
+	Logf("Start polling, timeout is %v seconds", maxSeconds)
+
+	//How long until the FIRST transaction...
+	startupTimeout := time.After(time.Duration(startupTimeoutSeconds) * time.Second)
+
+	//How long we will wait until we reach the nth (ie. 500th) transaction
+	transactionsCompleteTimeout := time.After(time.Duration(maxSeconds) * time.Second)
+	tick := time.Tick(2 * time.Second)
+	var ready = false
+
+	Logf("Now waiting %v seconds to see progress (transactions  > 3)", startupTimeoutSeconds)
+T:
+	for {
+		select {
+		case <-transactionsCompleteTimeout:
+			//i.e. if we dont have n*1000 transactions, fail the test.
+			Logf("Timeout %v reached. Breaking!", tick)
+			break T
+		case <-startupTimeout:
+			err, totalTransactions = readTransactions(c, ns)
+			Logf("Timeout %v reached.  Checking if transactions have occured.", startupTimeout)
+			//If we don't have 3 transactions, fail the test.
+			if err != nil {
+				Logf("Failed : Error %v", err)
+				break T
+			}
+			if totalTransactions < 3 {
+				break T
+			}
+			ready = true
+		case <-tick:
+			//Pass if we've collected enough transactions
+			err, totalTransactions = readTransactions(c, ns)
+			if ready {
+				Expect(err).NotTo(HaveOccurred())
+			}
+			if totalTransactions > minExpected {
+				break T
+			}
+			Logf("%v == %v total petstore transactions stored into redis. ==", time.Now(), totalTransactions)
+		}
+	}
+
+	//Finally! We should have exceeded the min_expected num of transactions.
+	//If this fails, but there are transactions being created, we may need to recalibrate
+	//the min_expected value - or else - your cluster is broken/slow !
+	Ω(totalTransactions).Should(BeNumerically(">", minExpected))
+}
+
+var _ = Describe("K8PetStore (succesfull launching and transactions in the k8petstore application)", func() {
+
+	var namespace *api.Namespace = nil
+	var ns string = ""
+
+	//This must run on a machine which is in the kube-proxy ring.  Otherwise, the publicIP binding will fail.
+	//The IP Below ~ letter->number cipher for k8petstore (most likely it won't be bound by any other process)
+	var ip = "165.201.92.15"
```

Comment:
This IP isn't needed for NodePorts, but it is a parameter we keep in the test as a parameter which can be used by the testing function.  Since the only running test function doesn't use the IP at all, I can delete the cipher entirely for now I guess to simplify things.